### PR TITLE
--Managed Container refactor to support Metadata Mediator

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR refactors the ManagedContainers to more appropriately support the upcoming MetadataMediator/DatasetAttributes functionality.  The following important changes were made : 

- Unused refs to ResourceManager were removed.
- Functions that took JSON documents as arguments now take generic JSON objects so they can more easily consume nested JSON cells, which will be present in upcoming configuration files for datasets, lighting layouts, and more.
- The non-template-specific code for the ManagedContainer class was moved to a non-templated abstract base class, for stability and to minimize code bloat.
- Substantial portion of create functionality was aggregated into base classes/class templates for the AttributesManagers.

Furthermore, 

## How Has This Been Tested
C++ and python tests

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: examples/tutorials/colabs/rigid_object_tutorial.ipynb [hunk 1]
```diff
@@ -169,7 +169,7 @@
     "    # [basics]\n",
     "\n",
     "    # load some object templates from configuration files\n",
-    "    sphere_template_id = obj_templates_mgr.load_object_configs(\n",
+    "    sphere_template_id = obj_templates_mgr.load_configs(\n",
```
Comment (jturner65) [inline examples/tutorials/colabs/rigid_object_tutorial.ipynb:172]: Note : "load_object_configs" still works, but is being deprecated, and will be removed at some point in the future

File: src/esp/metadata/managers/AbstractObjectAttributesManagerBase.h [hunk 2]
```diff
@@ -36,30 +36,71 @@ class AbstractObjectAttributesManager : public AttributesManager<T> {
...
   //======== Common JSON import functions ========
 
   /**
-   * @brief Create either an object or a stage attributes from a json config.
-   * Since both object attributes and stage attributes inherit from @ref
-   * esp::metadata::attributes::AbstractObjectAttributes, the functionality to
-   * populate these fields from json can be shared.  Also will will populate
-   * render mesh and collision mesh handles in object and stage attributes with
-   * value(s) specified in json.  If one is blank will use other for both.
+   * @brief Popualte a existing @ref
```
Comment (aclegg3) [inline src/esp/metadata/managers/AbstractObjectAttributesManagerBase.h:89]:
```suggestion
   * @brief Populate an existing @ref
```

File: src/esp/metadata/managers/AbstractObjectAttributesManagerBase.h [hunk 3]
```diff
@@ -36,30 +36,71 @@ class AbstractObjectAttributesManager : public AttributesManager<T> {
...
+   * metadata::attributes::AbstractObjectAttributes from a JSON config.  Also
+   * will will populate render mesh and collision mesh handles with value(s)
```
Comment (aclegg3) [inline src/esp/metadata/managers/AbstractObjectAttributesManagerBase.h:91]:
```suggestion
   * will populate render mesh and collision mesh handles with value(s)
```

**Review by aclegg3 — APPROVED** [review]
Looks good. Thanks for breaking this out. 👍

**Pull request merged**